google/cloud/aiplatform/models.py

```diff
@@ -1023,6 +1024,20 @@ def undeploy(
                     "Sum of all traffic within traffic split needs to be 100."
                 )
 
+        elif (
+            len(self.traffic_split) > 1  # Two or more models deployed to Endpoint
```

Comment:
Shouldn't this check for the case where all other deployed models' traffic split sums to 0? Or at least that's how I understood the bug.

Reply:
Also, shouldn't `undeploy_all` sort the models so that the lowest traffic split, potentially 0%, is undeployed first so we don't misconfigure the endpoint.

Reply:
You're right, this covers a different case:

```py
ep.traffic_split
# { 'm1': 35, 'm2': 65 }

ep.undeploy('m1'). # Raises error about "would leave the remaining traffic split at 65%"
```

shall I leave this case in while adding the following two changes:

```py
ep.traffic_split
# { 'm1': 0, 'm2': 0, 'm3':100 }

ep.undeploy('m3'). # Raises error about needing to undeploy 'm1', 'm2' first ​
```

```py
ep.traffic_split
# { 'm1': 0, 'm2': 100, 'm3':0 }


ep.undeploy_all()  # Undeploys 'm1', 'm3' first, then 'm2'
```